saltedge-compliance-connector-java-v2/src/main/java/com/saltedge/connector/example/controllers/consent/ConsentBaseController.java

```diff
@@ -32,66 +32,63 @@
 
 abstract public class ConsentBaseController extends BaseController {
 
-  protected ModelAndView onAisSuccess(
-      @NotEmpty String sessionSecret,
-      @NotEmpty String userId,
-      ProviderConsents consents
-  ) {
-    String returnToUrl = connectorCallbackService.onAccountInformationAuthorizationSuccess(
-        sessionSecret,
-        userId,
-        KeyTools.generateToken(32),
-        consents
-    );
-    if (returnToUrl == null) return redirectToAccountsAuth(sessionSecret, userId);
-    else return new ModelAndView("redirect:" + returnToUrl);
-  }
+    protected ModelAndView onAisSuccess(
+            @NotEmpty String sessionSecret,
+            @NotEmpty String userId,
+            ProviderConsents consents
+    ) {
+        String returnToUrl = connectorCallbackService.onAccountInformationAuthorizationSuccess(
+                sessionSecret,
+                userId,
+                KeyTools.generateToken(32),
+                consents
+        );
+        if (returnToUrl == null) return redirectToAccountsAuth(sessionSecret, userId);
+        else return new ModelAndView("redirect:" + returnToUrl);
+    }
+
+    protected ModelAndView onAisDenied(@NotEmpty String sessionSecret) {
+        String returnToUrl = connectorCallbackService.onAccountInformationAuthorizationFail(sessionSecret);
+        if (returnToUrl == null) return redirectToAccountsAuth(sessionSecret, null);
+        else return new ModelAndView("redirect:" + returnToUrl);
+    }
 
-  protected ModelAndView onAisDenied(@NotEmpty String sessionSecret) {
-    String returnToUrl = connectorCallbackService.onAccountInformationAuthorizationFail(sessionSecret);
-    if (returnToUrl == null) return redirectToAccountsAuth(sessionSecret, null);
-    else return new ModelAndView("redirect:" + returnToUrl);
-  }
+    protected ModelAndView onPisDenied(long paymentId) {
+        PaymentEntity payment = paymentsRepository.findById(paymentId).orElse(null);
```

Comment:
You can use the map function to avoid additional checks:
`return paymentsRepository.findById(paymentId).map(payment -> {`

Reply:
I don't understand how we can use `map` here.